src/msg/async/AsyncConnection.cc

```diff
@@ -730,7 +730,16 @@ void AsyncConnection::process()
 
             ldout(async_msgr->cct, 20) << __func__ << " got front " << front.length() << dendl;
           }
-          state = STATE_OPEN_MESSAGE_READ_MIDDLE;
+          if (current_header.middle_len) {
```

Comment:
actually, I expect we remove break statement and flow down the state, I think it will be cheap too.